hs/src/Reach/Verify/SMT.hs

```diff
@@ -231,9 +232,12 @@ smtVar :: DLVar -> App String
 smtVar dv@(DLVar _ _ _ i) = do
   pvs <- ctxt_primed_vars <$> ask
   let mp = case elem dv pvs of
-        True -> "p"
-        False -> ""
-  return $ "v" ++ show i ++ mp
+            True -> "p"
+            False -> ""
+  let name = "v" ++ show i ++ mp
+  v2dv <- ctxt_v_to_dv <$> ask
+  liftIO $ modifyIORef v2dv $ M.insert name dv
```

Comment:
This map, `v2dv`, is going to have duplicates in it and you're going to arbitrarily select the last one that was seen. I think you should try to come up with an example where the same values gets multiple names and try to see if you can get bad output. This then is a place to insert that "most specific" logic we talked about.